grammars/javascript.cson

```diff
@@ -17,8 +17,162 @@
 'name': 'JavaScript'
 'patterns': [
   {
+    'include': '#switch_statement'
+  }
+  {
+    # foo: function() { … }
+    'begin': '''(?x)
+      (\\b([a-zA-Z_?.$][\\w?.$]*))
+      \\s*(:)\\s*
+      \\b(?:(async)(?:\\s+))?
+      (function\\*?)
+      \\s*(\\*?)\\s*
+      ([a-zA-Z_?$][\\w?$]*)?
```

Comment:
I don't think this line is correct either, as object-literal functions have to be anonymous.